docs/contributor-guide.md

```diff
@@ -30,22 +30,181 @@ You can see which steps will be performed by running 'dh binary --no-act' in the
 
 In our case, we perform the following operations:
 - retrieve the user configuration using debconf
-- download the binaries
+- download the binaries for host architecture
 - add a user in the system that will run the services
+- copy identity to `/etc/storagenode/identity` if none
+- setup storagenode if it hasn't been setup yet
+- change ownership of `/var/lib/storagenode` and `/etc/storagenode` directories
 
 ### `prerm`
-This file contains the steps to be run when removing the package.
+This file contains the steps to be run before removing the package.
+
+### `postrm`
+This file contains the steps to be run after removing the package. Removes storagenode and storagenode-updater binaries.
 
 ### `storagenode.service` and `storagenode-updater.service`
 These are the services that will be installed and which will run the binaries.
 
+#### `storagenode` service unit file
+```service
+# This is a SystemD unit file for the Storage Node
+# To configure:
+# - Update the user and group that the service will run as (User & Group below)
+# - Ensure that the Storage Node binary is in /usr/local/bin and is named storagenode (or edit the ExecStart line
+#   below to reflect the name and location of your binary
+# - Ensure that you've run setup and have edited the configuration appropriately prior to starting the
+#   service with this script
+# To use:
+# - Place this file in /etc/systemd/system/ or wherever your SystemD unit files are stored
+# - Run systemctl daemon-reload
+# - To start run systemctl start storagenode
+
+[Unit]
+Description  = Storage Node service
+After        = syslog.target
+
+[Service]
+Type         = simple
+User         = storj-storagenode
+Group        = storj-storagenode
+ExecStart    = /var/lib/storagenode/storagenode run --config-dir "/etc/storagenode"
+Restart      = on-failure
+NotifyAccess = main
+
+[Install]
+Alias        = storagenode
+WantedBy     = multi-user.target
+```
+
+#### `storagenode-updater` service unit file
+```service
+# To configure:
+# - Update the user and group that the service will run as (User & Group below)
+# - Ensure that the Storage Node binary is in /usr/local/bin and is named storagenode (or edit the ExecStart line
+#   below to reflect the name and location of your binary
+# - Ensure that you've run setup and have edited the configuration appropriately prior to starting the
+#   service with this script
+# To use:
+# - Place this file in /etc/systemd/system/ or wherever your SystemD unit files are stored
+# - Run systemctl daemon-reload
+# - To start run systemctl start storagenode
+
+[Unit]
+Description  = Storage Node Updater service
+After        = syslog.target
+
+[Service]
+Type         = simple
+User         = storj-storagenode
+Group        = storj-storagenode
+ExecStart    = /var/lib/storagenode/storagenode-updater run --config-dir "/etc/storagenode" --binary-location "/var/lib/storagenode/storagenode"
```

Comment:
I see a problem with the binary-location. I am running 6 storage nodes. If they all use the same binary this also means they will all update at the same time. As a workaround I have now 6 binaries and 6 update services. That way the storage nodes can update one by one.

Any idea how we solve this issue?

Reply:
In the architecture review meeting we agreed that all of my nodes can update at the same time. I could run 1 updater and 6 storage nodes.